### PR TITLE
static-analysis

### PR DESCRIPTION
**Pull request by sokrahta**
* update sbt to 0.13.8
* replace socrata-cloudbees-sbt with socrata-sbt-plugins
* replace assembly merge strategy and exclude icky libraries
* replace Version generated source with BuildInfo
* fix some (now) fatal errors

### CODE REVIEW

File: project/BalboaAgent.scala [hunk 1]
```diff
@@ -4,28 +4,16 @@ import com.typesafe.sbt.packager.debian
...
 import sbt.Keys._
 import sbt._
-import sbtassembly.Plugin.AssemblyKeys._
 
 object BalboaAgent {
-  lazy val settings: Seq[Setting[_]] = BuildSettings.projectSettings(assembly = true) ++ Seq(
-    mainClass in assembly := Some("com.socrata.balboa.agent.BalboaAgent"),
+  lazy val settings: Seq[Setting[_]] = BuildSettings.projectSettings ++ Seq(
     libraryDependencies <++= scalaVersion { libraries(_) },
-    jarName in assembly := s"${name.value}-assembly.jar",
```
Comment (mhotan) [inline project/BalboaAgent.scala]: I need to verify this does not break the balboa agent path configuration.

**Comment by mhotan** [comment]
@sokrahta is there any benefit to having the entier build be defined within <project root>/build.sbt rather then <project root>/project/*scala files?

File: project/BalboaAgent.scala (hunk 1 above)
Comment (sokrahta) [inline project/BalboaAgent.scala]: The way other services run is with `<service>-assembly-<version>.jar`. If we can use this standard it would be great.

**Comment by sokrahta** [comment]
`/build.sbt` and `/project/Build.scala` are semantically equal.

**Comment by sokrahta** [comment]
@mhotan balboa-agent/debian:packageBin exercised and seem to be working.

**Comment by mhotan** [comment]
Thanks for the PR! :shipit:

File: project/BalboaAgent.scala [hunk 2]
```diff
@@ -1,31 +1,19 @@
...
 import sbt._
-import sbtassembly.Plugin.AssemblyKeys._
 
 object BalboaAgent {
-  lazy val settings: Seq[Setting[_]] = BuildSettings.projectSettings(assembly = true) ++ Seq(
-    mainClass in assembly := Some("com.socrata.balboa.agent.BalboaAgent"),
+  lazy val settings: Seq[Setting[_]] = BuildSettings.projectSettings ++ Seq(
+    mainClass in sbtassembly.AssemblyKeys.assembly := Some("com.socrata.balboa.agent.BalboaAgent"),
     libraryDependencies <++= scalaVersion { libraries(_) },
-    jarName in assembly := s"${name.value}-assembly.jar",
```
Comment (mhotan) [inline project/BalboaAgent.scala]: Ohh wait this is actuall important.  We will need to add the following block back.

Comment (sokrahta) [inline project/BalboaAgent.scala]: could we add a symlink to the versioned jar?

**Comment by sokrahta** [comment]
this is what the findbugs analysis is going to look like: https://docs.google.com/document/d/1jeBmpQXo9K8p8Ik24R00I6CLILrTQtsz22HGvR9ihkU

File: project/BalboaAgent.scala (hunk 2 above)
Comment (mhotan) [inline project/BalboaAgent.scala]: Yeah Ill just change the debian to a plain old assembly jar to coincide with the way we are doing things.

**Comment by mhotan** [comment]
A little this will reveal something about the build system that I don't understand and break one of our jenkins jobs.  :shipit: but @sokrahta I can I merge this when I get Balboa fully tested in AWS?

**Comment by sokrahta** [comment]
@mhotan yeah this is ready for merge now, thanks!

**Comment by sokrahta** [comment]
cheers :checkered_flag:

**Pull request merged**